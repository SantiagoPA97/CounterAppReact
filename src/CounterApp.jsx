import { useState, React } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ( { value } ) => {

    const [ counter, setCounter ] = useState(value);

    const handleAdd = (event) => {
        // console.log(event);
        // setCounter(counter + 1);
        setCounter((c) => c+1);
    }

    const handleOperation = ( value ) => {
        // console.log(event);
        // setCounter(counter + 1);
        setCounter((c) => c+value);
    }

    const handleSubstract = (event) => {
        // console.log(event);
        // setCounter(counter + 1);
        setCounter((c) => c-1);
    }

    const handleReset = (event) => {
        // console.log(event);
        // setCounter(counter + 1);
        setCounter((c) => value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>

            <button onClick={ () => handleOperation(1) }>+1</button>
            <button onClick={ () => handleOperation(-1) }>-1</button>
            <button aria-label="btn-reset" onClick={ handleReset }>Reset</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}
