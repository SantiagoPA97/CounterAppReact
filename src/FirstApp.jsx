import React from 'react';
import PropTypes from 'prop-types';

export const FirstApp = ( { title, subtitle, name } ) => {

    return (
        <>
            {/* <code> { JSON.stringify(newMessage) } </code> */}
            <h1>{ title }</h1>
            {/* <h2> { getResult() } </h2> */}
            <p>{ subtitle }</p>
            <p>{ name }</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

FirstApp.defaultProps = {
    // title: 'No hay titulo',
    subtitle: 'No hay subtitulo',
    name: 'No hay nombre'
}
