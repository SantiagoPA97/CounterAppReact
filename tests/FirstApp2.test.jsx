import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Tests in <FirstApp />', () => { 
   
    const title = 'Hola, soy Goku';
    const subTitle = 'Soy un subtitulo';

    test('should match with snapshot', () => { 
        const { container } = render(<FirstApp title={ title } />);
        expect(container).toMatchSnapshot();
    });

    test('should show the message "Hola, soy Goku"', () => {
        render(<FirstApp title={ title } />);
        expect(screen.getByText(title)).toBeTruthy();
        //screen.debug();
    });

    test('should show the title in h1 tag', () => {
        render(<FirstApp title={ title } />);
        expect(screen.getByRole('heading', { level: 1})).toBeTruthy();
        //screen.debug();
    });

    test('should show the subtitle sent in the props', () => {
        render(<FirstApp title={ title } subtitle={ subTitle } />);
        expect(screen.getAllByText(subTitle).length).toBe(1);
    });
    
 })