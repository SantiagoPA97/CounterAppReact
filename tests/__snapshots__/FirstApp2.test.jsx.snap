// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should match with snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay nombre
  </p>
</div>
`;
